mfhdf/test/tncvargetfill.c

```diff
@@ -260,6 +270,143 @@ test_1dim_multivars()
 #define VAR1D       "Variable 1D"
 #define VAREMPTY    "Variable EMPTY"
 #define VARDOZEN    "Variable Dozen Records"
+
+/*
+ * Helper function: Read and verify data of the variables with nc API
+ */
+static int
+read_verify_nc_api_multidims(void)
+{
+    long  start[3];
+    long  edges[3];
+    int   ncid;                        /* file id */
+    int   var1id, var2id, var3id;      /* variable ids */
+    long  dims[3];                     /* dimension size buffer */
+    int   rh_ndims;                    /* number of dims */
+    int   rh_dims[H4_MAX_VAR_DIMS];    /* variable shape */
+    char  varname[H4_MAX_NC_NAME];     /* variable name */
+    int16 outdata1D[DIM0];             /* 1-D data read back */
+    int16 outdata3D[DIM0][DIM1][DIM2]; /* 3-D data read back */
+    int   status   = 0;                /* returned by called functions */
+    int   num_errs = 0;                /* number of errors so far */
+
+    /* After the fourth data set (VARDOZEN) was added, the maximum number
+       of records became 12.  Thus, the results must be changed to reflect
+       the behavior in nc API. */
+    /* clang-format off */
+    int16 result3D[DIM0][DIM1][DIM2] =
+    {
+        {{300,  -3}, { -3,  -3}, { -3,  -3}},
+        {{301,  -3}, { -3,  -3}, { -3,  -3}},
+        {{302,  -3}, { -3,  -3}, { -3,  -3}},
+        {{303,  -3}, { -3,  -3}, { -3,  -3}},
+        {{ -3,  -3}, { -3,  -3}, { -3,  -3}},
+        {{ -3,  -3}, { -3,  -3}, { -3,  -3}},
+        {{ -3,  -3}, { -3,  -3}, { -3,  -3}},
+        {{800, 801}, {802, 803}, {804, 805}},
+        {{ -3,  -3}, { -3,  -3}, { -3,  -3}},
+        {{ -3,  -3}, { -3,  -3}, { -3,  -3}},
+        {{ -3,  -3}, { -3,  -3}, { -3,  -3}},
+        {{ -3,  -3}, { -3,  -3}, { -3,  -3}}
+    };
```

Comment:
@bmribler - Make sure I didn't do anything wrong here. I'm new to the HDF4 API but it looks like this was what was intended.